src/news-app.html

```diff
@@ -216,26 +213,28 @@
         // Safari. To workaround this on Chrome on iOS, we will use a
         // fixed viewport height in places where normally relying on "vh" unit and replace them with
         // custom property "--viewport-height".
-        var ua = navigator.userAgent;
-        var cMatch = navigator.userAgent.match(/Android.*Chrome[\/\s](\d+\.\d+)/);
+        let ua = navigator.userAgent;
+        let cMatch = navigator.userAgent.match(/Android.*Chrome[\/\s](\d+\.\d+)/);
         if (ua.match('CriOS') || (cMatch && cMatch[0] && cMatch[1] < 56)) {
           document.body.classList.add('fixed-viewport-height');
         }
-      },
 
-      attached: function() {
-        Polymer.RenderStatus.afterNextRender(this, function() {
-          this.listen(window, 'online', '_notifyNetworkStatus');
-          this.listen(window, 'offline', '_notifyNetworkStatus');
+        Polymer.RenderStatus.afterNextRender(this, () => {
+          window.addEventListener('online', (e)=>this._notifyNetworkStatus(e));
+          window.addEventListener('offline', (e)=>this._notifyNetworkStatus(e));
+          this.addEventListener('refresh-data', (e)=>this._refreshData(e));
         });
-      },
+      }
 
-      detached: function() {
-        this.unlisten(window, 'online', '_notifyNetworkStatus');
-        this.unlisten(window, 'offline', '_notifyNetworkStatus');
-      },
+      connectedCallback() {
```

Comment:
You could set the property in `ready` or into the `Polymer.RenderStatus.afterNextRender` above in case you want to avoid declaring `connectedCallback`